### PR TITLE
cli: major refactor

### PR DESCRIPTION
**Pull request by steverusso**
fixes: #1349
fixes: #1357
fixes: #1388
fixes: #1552
fixes: #1553

### CODE REVIEW

File: clients/cli/Cargo.toml [hunk 1]
```diff
@@ -1,18 +1,10 @@
 [package]
-name = "lockbook-cli"
-version = "0.5.10"
+name = "cli2"
+version = "0.1.0"
 edition = "2021"
 
-[[bin]]
-name = "lockbook"
-path = "src/main.rs"
```
Comment (Parth) [inline clients/cli/Cargo.toml:8]: probably do want the og values here

File: clients/cli/src/account.rs [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+    std::io::stdout().flush()?;
+
+    let username = match maybe_username {
+        Some(uname) => uname,
+        None => {
+            let mut uname = String::new();
+            std::io::stdin()
+                .read_line(&mut uname)
+                .expect("failed to read from stdin");
+            uname.retain(|c| c != '\n' && c != '\r');
```
Comment (Parth) [inline clients/cli/src/account.rs:54]: don't feel strongly, but maybe we want to chuck this business in a fn

File: clients/cli/src/list.rs [hunk 3]
```diff
@@ -1,44 +1,209 @@
...
-        if ids {
-            println!("{}: {}", get_by_path(core, &path)?, path)
+use crate::CliError;
+use crate::ID_PREFIX_LEN;
+
+#[derive(Parser, Debug)]
+pub struct ListArgs {
+    /// include all children of the given directory, recursively
+    #[clap(short, long)]
+    recursive: bool,
```
Comment (Parth) [inline clients/cli/src/list.rs:17]:
```suggestion
    recursive: bool,
    
```

File: clients/cli/src/list.rs [hunk 4]
```diff
@@ -1,44 +1,209 @@
...
+    /// include more info (such as the file ID)
+    #[clap(short, long)]
+    long: bool,
```
Comment (Parth) [inline clients/cli/src/list.rs:20]:
```suggestion
    long: bool,
    
```
and so on?

File: clients/cli/src/main.rs [hunk 5]
```diff
@@ -1,265 +1,245 @@
...
+            PushDocument(name) => println!("pushing: {}", name),
+        };
+    })))?;
+    Ok(())
 }
 
-fn exit_with(err: CliError) -> ! {
-    err.print();
-    std::process::exit(err.code as i32)
+fn create(core: &Core, path: &str) -> Result<(), CliError> {
```
Comment (Parth) [inline clients/cli/src/main.rs:189]: feels weird that some of the fns are here, no?

**Review by Parth — COMMENTED** [review]
looks sick, very excited

**Comment by Parth** [comment]
Feel free to paste into your desc:

seemingly done:

```
fixes: #1349
fixes: #1357
fixes: #1388
fixes: #1552
fixes: #1553
```

not done yet:

```
fixes: #878
```

File: clients/cli/Cargo.toml (hunk 1 above)
Comment (steverusso) [reply clients/cli/Cargo.toml:8]: oops

File: clients/cli/src/main.rs (hunk 5 above)
Comment (steverusso) [reply clients/cli/src/main.rs:189]: Not weirder than splitting them up into many small files imo. Your call

File: clients/cli/src/account.rs (hunk 2 above)
Comment (steverusso) [reply clients/cli/src/account.rs:54]: Which business, just reading the username from stdin?

File: clients/cli/Cargo.toml (hunk 1 above)
Comment (steverusso) [reply clients/cli/Cargo.toml]: https://github.com/lockbook/lockbook/pull/1561/commits/f153fc09e0e028af588cfd7b232af04f02a757ff

File: clients/cli/src/list.rs (hunk 3 above)
Comment (steverusso) [reply clients/cli/src/list.rs:17]: https://github.com/lockbook/lockbook/pull/1561/commits/40ade61b05a12dc9a3dd6c32aa8fe73ecfb399ac

File: clients/cli/src/list.rs (hunk 4 above)
Comment (steverusso) [reply clients/cli/src/list.rs:21]: https://github.com/lockbook/lockbook/pull/1561/commits/40ade61b05a12dc9a3dd6c32aa8fe73ecfb399ac

File: clients/cli/src/account.rs (hunk 2 above)
Comment (Parth) [reply clients/cli/src/account.rs:54]: yeah looks like we do have that in the main.rs

Comment (steverusso) [reply clients/cli/src/account.rs:54]: Ah, good call

Comment (steverusso) [reply clients/cli/src/account.rs]: https://github.com/lockbook/lockbook/pull/1561/commits/7539bfd9738125308c984645cf3ff94ddc429c8c
(also fixed a future bug lol the prompt was there before I even checked for the optional username arg)

**Review by Parth — APPROVED** [review]

**Pull request merged**